### PR TITLE
Change the wording layout to groupby

### PR DESCRIPTION
**Pull request by pmyip3**
# Description 
As title 

# Type of change 
- [ ] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 

# Problem
The word layout is a bit confusing 

# Solution description
Change the relevant files in api and client
** Haven't changed related files in scholly as the new scholly layouts haven't considered the saved sorting order from editor site, will fix this in a separate ticket. 

# Tests 
All unit tests pass? N/A

Builds successfully? Yes

Attach screenshot of the expected result (e.g. UI of the website/ postman api call result/ database changes)
image
image

Provide steps in detail for reviewers to reproduce the expected result of the code changes.
Change the group by option, it should work the same as the layout

# Relevant document/ link (if any) 
Any related documents could help reviewers to review this PR 

# Risk
Low

### CODE REVIEW

File: scholly/base/src/shared/config/publications.js [hunk 1]
```diff
@@ -9,19 +9,16 @@ const categoryType = {
   Book: 'Book',
   Other: 'Other',
 };
-const layoutOptions = {
-  ALL_PUBLICATION: 'All Publication',
-  BY_CATEGORY: 'By Category',
+const groupByOptions = {
+  NONE: 'None',
+  CATEGORY: 'Category',
```
Comment (vgoh1) [inline scholly/base/src/shared/config/publications.js:14]: Just wondering, do we need this in scholly/base if they're not being used? Or just for consistency? I think the only ones being used in the layouts are the page sizes.

**Comment by vgoh1** [comment]
I think there might be an edge case for existing teams in this branch where you log in and it just shows `Group By:` without an option selected, but once I chose an option and saved it was fine. I also see that you renamed the existing docs in the DB so I'm not sure what could've caused it, but it doesn't break the site so it shouldn't be too much of an issue.

File: scholly/base/src/shared/config/publications.js (hunk 1 above)
Comment (pmyip3) [reply scholly/base/src/shared/config/publications.js:14]: I was planning to change it to a separate ticket, but Scholly will need that for my next ticket to display the publication based on the sorting/ layout options.

**Comment by pmyip3** [comment]
> s

It's because the options data haven't been returned from the API call. Previously, I was putting the default sorting/ layout in the redux state so it displays the default one instead of being blank, but I don't find it nice as the options will change automatically after the option data is returned from API call, so I just removed the default sorting/ layout in the redux state. If we are going to implement a loading page, then it should be fine no matter which way we use.

**Review by vgoh1 — APPROVED** [review]
Makes sense, approval from me.

**Review by aksh0001 — APPROVED** [review]

**Review by jfan0010 — APPROVED** [review]
Works on mine

**Pull request merged**